Clarification and plugin-inclusion (#12)
SAP-docs/btp-application-autoscaler: docs/cf-cli-plugin.md

# Application Autoscaler Plugin for the Cloud Foundry CLI
The Application Autoscaler offers a plugin for the Cloud Foundry Command Line Interface CLI. The plugin enables users to:
  * attach or detach scaling policies to apps
  * view attached scaling policies and view the scaling histories

>   * view the metrics sent by the app

Author: I think there is no context here specifying which app is meant by “the app”. I would be less specific here. As an example:
```suggestion
  * view the metrics sent by apps
```

> # Plug-in for the “Cloud Foundry CLI”

Author: Writing “plugin” is an orthographic error, see: <https://dictionary.cambridge.org/dictionary/english/plug-in?q=plugin>

> “Application Autoscaler” offers a plug-in for the “Cloud Foundry CLI”. With it users can attach or detach scaling-policies to their apps. They can view them and view the scaling-history. And they can view the metrics send by their apps.

Reviewer: As they can also view built-in metrics I would phrase this sentence a bit more general 
```suggestion
“Application Autoscaler” offers a plug-in for the “Cloud Foundry CLI”. With it users can attach or detach scaling-policies to their apps. They can view them and view the scaling-history. And they can view the metrics the scaling policies are based on.
```

> “Application Autoscaler” offers a plug-in for the “Cloud Foundry CLI”. With it users can attach or detach scaling-policies to their apps. They can view them and view the scaling-history. And they can view the metrics send by their apps.

Author: Of course, we can as well change now that output of the cli. What do you think? @silvestre

> “Application Autoscaler” offers a plug-in for the “Cloud Foundry CLI”. With it users can attach or detach scaling-policies to their apps. They can view them and view the scaling-history. And they can view the metrics send by their apps.

Author: Btw. I have set a non-breaking whitespace in `Application Autoscaler` for better readability because then both words will be enforced to stay at the same line. (Personally I prefer the more visible `Application-Autoscaler` which also shows to humans that it belongs together, but I don't want to touch that here.)

> This is useful to get some insights on how currently applied scaling-policies work and if changes are reasonable. It is as well useful for debugging. A common case is, when [custom-metrics](<../defining-a-custom-metric-87e657e.md>) are introduced to scale an app and the developers want to check, if the send metrics are correctly processed by Autoscaler and why up-/down-scaling happens or does not happen.

Reviewer: The "Application Autoscaler" only scales horizontally, so we should use the related verbs:

```suggestion
This is useful to get some insights on how currently applied scaling-policies work and if changes are reasonable. It is as well useful for debugging. A common case is, when [custom-metrics](<../defining-a-custom-metric-87e657e.md>) are introduced to scale an app and the developers want to check, if the send metrics are correctly processed by Autoscaler and why scaling out or in happens or does not happen.
```

Author: Great! I didn't knew that. With “up” and “down” I actually meant the number of instances (value of a real-numerical variable goes go up or down) but this is a technical term. <https://en.wikipedia.org/wiki/Scalability#HORIZONTAL-SCALING>

> 

Reviewer: I don't think our help pages need to be completely self-contained, so for details we can provide a link, but the main content should be in help.sap.com.

|------------------------------------------------------------------|------------------------------------------------|
| [autoscaling-api, asa](#cf-autoscaling-api)                      | Set or view AutoScaler service API endpoint    |
| [autoscaling-policy, asp](#cf-autoscaling-policy)                | Retrieve the scaling policy of an application  |
| [attach-autoscaling-policy, aasp](#cf-attach-autoscaling-policy) | Attach a scaling policy to an application      |
| [detach-autoscaling-policy, dasp](#cf-detach-autoscaling-policy) | Detach the scaling policy from an application  |
| [autoscaling-metrics, asm](#cf-autoscaling-metrics)              | Retrieve the metrics of an application         |
| [autoscaling-history, ash](#cf-autoscaling-history)              | Retrieve the scaling history of an application |

## Command usage
### `cf autoscaling-api`
Set or view the API-endpoint of the service Autoscaler to be used by this CLI. If the CF API endpoint is <https://api.example.com>, then typically the Autoscaler-API-endpoint will be https://autoscaler.example.com. Check the manifest when Autoscaler is deployed to get the API-endpoint.

> ### Note:

> > This command is usually not needed when working with BTP-Cloud-Foundry.

Reviewer: See https://brand.sap.com/naming-center/search?keywords=cloud%2520foundry&status=APPROVED
```suggestion
> This command is usually not needed when working with SAP BTP, Cloud Foundry environment.
```

Author: Didn't know this as well.
